core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -131,8 +137,8 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
                     }
                 }
             }
+            used.add(c);
```

Comment:
there's a chance `c` is going to be `null` here right? Granted a node with no executor should be quite rare I guess but well :).
Or maybe the passed `used` container accepts null values?